contracts/liquidityMining/KyberFairLaunch.sol

```diff
@@ -113,69 +131,76 @@ contract KyberFairLaunch is IKyberFairLaunch, PermissionAdmin, ReentrancyGuard {
 
   constructor(
     address _admin,
-    IERC20Ext _rewardToken,
     IKyberRewardLocker _rewardLocker
   ) PermissionAdmin(_admin) {
-    rewardToken = _rewardToken;
     rewardLocker = _rewardLocker;
-
-    _rewardToken.safeApprove(address(_rewardLocker), type(uint256).max);
   }
 
+  receive() external payable {}
+
   /**
-   * @dev allow admin to withdraw only reward token
+   * @dev allow admin to withdraw in case of anything happens
    */
-  function adminWithdraw(uint256 amount) external onlyAdmin {
-    rewardToken.safeTransfer(msg.sender, amount);
+  function adminWithdraw(IERC20Ext token, uint256 amount) external onlyAdmin {
+    if (token == IERC20Ext(0)) {
+      (bool success, ) = msg.sender.call{ value: amount }('');
+      require(success, 'transfer reward token failed');
+    } else {
+      token.safeTransfer(msg.sender, amount);
```

Comment:
It is important to restrict admin can not withdraw `stakeToken`

Reply:
May be add a TODO here is fine

Reply:
well, it is tricky now, since an admin can add whatever reward tokens, thus, if the admin wants to withdraw LP tokens, can just add a LP tokens as a reward token for some pools

Reply:
All LMs I know of use native tokens as reward tokens... I think it's unlikely for LP tokens to be reward tokens